rfc-171-remove-legacy-browser-js-support.md

```diff
@@ -0,0 +1,106 @@
+
+We should also adopt the Design System's recommendations for [browser support](https://frontend.design-system.service.gov.uk/browser-support/). Legacy browsers would be tested initially to ensure that they do not execute JS, then treated as Grade X browsers as recommended.
+
+### Alternative options
+
+For clarity here are the other options available to us, which we are not proposing.
+
+#### Don't upgrade to govuk-frontend v5
+
+This isn't a valid option. We need to keep up to date with `govuk-frontend` and all the functional and accessibility improvements this will bring.
+
+#### Upgrade, but don't use modern JS
+
+`govuk-frontend` includes an [option for not using ES modules](https://frontend.design-system.service.gov.uk/import-javascript/) but this will result in larger JS files, which would reduce the performance of GOV.UK. Support for this option may not continue indefinitely.
+
+Our initial upgrade to version 5.1 is planned to use this option but ultimately we should move away from it.
+
+#### Upgrade, but ignore the needs of legacy browsers
+
+Although users of legacy browsers represent only a very small percentage of the overall traffic on GOV.UK we have a responsibility to ensure that all visitors to the site have access to the information they need. We follow the principle of progressive enhancement, which means that the site should be usable from only the HTML - the addition of CSS and JS should be considered enhancements to improve the site, not an essential part of the system.
+
+If we ignore the needs of legacy browsers entirely we risk making the user experience for people with those browsers worse.
+
+#### Upgrade and make all the things work in legacy browsers
+
+It isn't possible to make legacy browsers understand modern JS, but it might be possible to write all of our code using modern JS and use a tool such as Babel to transpile it into older syntax.
+
+While this might be possible, we do not currently have this as part of our build pipeline and considerable effort would be required to introduce it. This would also significantly increase the size of our JS, as it would add additional code to polyfill modern functionality - we would be making GOV.UK slower for everyone for the benefit of a small minority. We would also lose the ability to embrace modern JS in the future.
+
+#### Upgrade, keep some things working for legacy browsers
+
+While many of our components would rely on ES modules JS from `govuk-frontend`, a lot of our other code does not, and could in theory still work. This would mean legacy browsers would experience some but not all of the JS enhancement of the site.
+
+While this might be possible, the developer burden of maintaining these two parts of the JS would be unsustainable, for little reward.
```

Comment:
Agreed, I think this also makes it difficult to communicate what users can expect to function as before by having partial/mixed support for IE11 (Grade X Browsers)